packages/server/modules/core/errors/base.js

```diff
@@ -0,0 +1,80 @@
+const VError = require('verror')
+
+/**
+ * Base application error (don't use directly, treat it as abstract). Built on top of `verror` so that you can
+ * chain errors (e.cause is the previous error) and also add arbitrary metadata using the `info` option.
+ *
+ * This allows for much nicer error handling & monitoring
+ */
+class BaseError extends VError {
```

Comment:
Added VError as an abstract base error class (adjusted apollo server response error formatting to output these kinds of errors as well).

VError is very nice, because it allows you to add specify a `cause` Error in the Error constructor. So you can chain errors this way (e.g. when you catch an error and then throw a new one) and get full stack traces for all errors in the chain and a concatenated error message - very useful for debugging!

Additionally you can add any kind of arbitrary metadata when invoking the constructor through the `info` options property. Monitoring tools like Sentry rely on grouping errors by their name, and with this feature you no longer have to build dynamic error names w/ relevant metadata (e.g. UserID), you can just add it to the metadata that can easily be sent to Sentry as well.

```js
// The message would be: "Couldn't create user: Lost connection to database"
try {
  await createUser(userInfo);
} catch (e) {
  throw new VError("Couldn't create user, {
     cause: e,
     info: {userId: userInfo.id}
  }
})
}
```